src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithPlainTextCommentFilter.java

```diff
@@ -0,0 +1,339 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.filters;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+import com.puppycrawl.tools.checkstyle.api.AuditEvent;
+import com.puppycrawl.tools.checkstyle.api.AutomaticBean;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+import com.puppycrawl.tools.checkstyle.api.Filter;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+
+/**
+ * <p>
+ *     A filter that uses comments to suppress audit events.
+ *     The filter can be used only to suppress audit events received from
+ *     {@link com.puppycrawl.tools.checkstyle.api.FileSetCheck} checks.
+ *     SuppressWithPlainTextCommentFilter knows nothing about AST,
+ *     it treats only plain text comments and extracts the information required for suppression from
+ *     the plain text comments. Currently the filter supports only single line comments.
+ * </p>
+ * <p>
+ *     Rationale:
+ *     Sometimes there are legitimate reasons for violating a check. When
+ *     this is a matter of the code in question and not personal
+ *     preference, the best place to override the policy is in the code
+ *     itself.  Semi-structured comments can be associated with the check.
+ *     This is sometimes superior to a separate suppressions file, which
+ *     must be kept up-to-date as the source file is edited.
+ * </p>
+ * @author Andrei Selkin
+ */
+public class SuppressWithPlainTextCommentFilter extends AutomaticBean implements Filter {
```

Comment:
`[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] ....checkstyle/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithPlainTextCommentFilter.java:[59,8] com.puppycrawl.tools.checkstyle.filters.SuppressWithPlainTextCommentFilter is not abstract and does not override abstract method finishLocalSetup() in com.puppycrawl.tools.checkstyle.api.AutomaticBean
`

@MEZk , please rebase.

Reply:
Rebased